telemetry/decision.go

```diff
@@ -0,0 +1,79 @@
+package telemetry
+
+import (
+	"log"
+	"os"
+	"path"
+	"path/filepath"
+)
+
+const OPT_OUT_FILE = "telemetry_opt_out"
+const OPT_IN_FILE = "telemetry_opt_in"
+
+func optOut() bool {
+	// We check that the user has made a decision so that we can have a default setting for robots
+	return userOptedOut() || envOptOut() || localOptOut() || made()
+}
+
+func made() bool {
+	return userOptedIn() || userOptedOut()
+}
+
+func userOptedOut() bool {
+	file := path.Join(home(), OPT_OUT_FILE)
```

Comment:
instead of writing multiple func to check file stat create one generic and use that.

Reply:
Done